astrocut/cube_cut.py

```diff
@@ -424,8 +431,21 @@ def _get_cutout(self, transposed_cube, verbose=True):
             ymax = ymax_cube       
         
         # Doing the cutout
-        cutout = transposed_cube[xmin:xmax, ymin:ymax, :, :]
-    
+        if threads == "auto" or threads > 1:
```

Comment:
One more thing: The multithread logic only works when the input cube is on the cloud, correct? What happens when a locally-stored cube gets fed, and `threads` is set to >1? Perhaps we should add an extra conditional here that ensures`transposed_cube` contains `stpubdata` in the path string, or something like that.

Reply:
> What happens when a locally-stored cube gets fed, and threads is set to >1? 

threads is reassigned to 1 if it doesn't start with s3

> Perhaps we should add an extra conditional here that ensures the input cube contains stpubdata in the path string, or something like that.

I don't think I understand, why would we do that?

Reply:
> threads is reassigned to 1 if the cube file doesn't start with s3

Ah okay, I see that piece of the code now. Nvm, thanks!